sdk/python/tests/unit/api/test_search_api.py

```diff
@@ -0,0 +1,2407 @@
+            if project:
+                services_by_project[project] = service
+
+        # Should have user_service in both project_a and project_b
+        assert "project_a" in services_by_project
+        assert "project_b" in services_by_project
+
+        # Verify they have different descriptions (different contexts)
+        desc_a = services_by_project["project_a"]["description"]
+        desc_b = services_by_project["project_b"]["description"]
+        assert desc_a != desc_b
+        assert "ride sharing" in desc_a
+        assert "food delivery" in desc_b
+
+    def test_all_resource_types_always_searched(self, search_test_app):
+        """Test that all resource types are always included in search results"""
+        response = search_test_app.get("/search?query=")
+        assert response.status_code == 200
+
+        data = response.json()
+        results = data["results"]
+
+        # Get all resource types returned
+        returned_types = set(result["type"] for result in results)
+
+        # Should include all expected resource types (including new 'feature' type)
+        expected_types = {
+            "entity",
+            "featureView",
+            "feature",
+            "featureService",
+            "dataSource",
+            "savedDataset",
+        }
+
+        # All expected types should be present (or at least no filtering happening)
+        # Note: Some types might not exist in test data, but if they do exist, they should all be returned
+        available_types_in_data = expected_types.intersection(returned_types)
+        assert len(available_types_in_data) >= 4, (
+            f"Expected multiple resource types in results, but only got {returned_types}. "
+            "All available resource types should be searched."
+        )
+
+    def test_features_as_individual_search_results(self, search_test_app):
+        """Test that individual features appear as separate search results"""
+        response = search_test_app.get("/search?query=")
+        assert response.status_code == 200
+
+        data = response.json()
+        results = data["results"]
+
+        # Find feature results
+        feature_results = [result for result in results if result["type"] == "feature"]
+
+        # Should have individual features in results
+        assert len(feature_results) > 0, (
+            "Expected individual features to appear in search results, but found none"
+        )
+
+        # Verify feature result structure
+        for feature_result in feature_results:
+            # Check required fields
+            assert "type" in feature_result
+            assert "name" in feature_result
+            assert "description" in feature_result
+            assert "project" in feature_result
+
+            # Verify values
+            assert feature_result["type"] == "feature"
+            assert isinstance(feature_result["name"], str)
+
+    def test_feature_search_by_name(self, search_test_app):
```

Comment:
- Should not be part of MultiProject Test class. Move to singular.
- Repeating test or club with existing.